syntax = "proto3";

import "base.proto";

package chat_proto;

message ChatMessage{
	MessageType type = 1;
	bytes message_body = 2;
}

message LoginRequest{
	uint32 user_id = 1;
	bytes user_name = 2;
	bytes password = 3;
}

message LoginResponse{
	Error error_status = 1;
}
message LoginoutRequest{
	uint32 user_id = 1;
}
message RegisterRequest{
	bytes user_name = 1;
	bytes password = 2;
}

message RegisterResponse{
	Error error_status = 1;
	uint32 user_id = 2;
}

message OneChatRequest{
	uint32 to_id = 1;
	bytes chat_message = 2;
}
message AddFriendRequest{
	uint32 user_id = 1;
	uint32 friend_id = 2;
	bytes verify_message = 3;
}
message AddFriendResponse{
	uint32 user_id = 1;
	uint32 friend_id = 2;
	bool pass = 3;
	bytes reject_message = 4;
}

message CreateGroupRequest{
	uint32 user_id = 1;
	bytes group_name = 2;
	bytes group_description = 3;
}
message CreateGroupResponse{
	Error error_status = 1;
	uint32 group_id = 2;
}
message AddGroupRequest{
	uint32 user_id = 1;
	uint32 group_id = 2;
}
message AddGroupResponse{
	bool pass = 1;
	bytes reject_message = 2;
}
message GroupChatRequest{
	uint32 user_id = 1;
	uint32 group_id = 2;
	bytes group_chat_message = 3;
}